### DISCUSSION TITLE
cmake Platform is somehow being set

### DISCUSSION THREAD

**Discussion by George-Gi** [microsoft/vscode-cmake-tools#2892] [Q&A]
I am getting the following error:
```
[cmake]     MinGW Makefiles
[cmake] 
[cmake]   does not support platform specification, but platform
[cmake] 
[cmake]     x64
[cmake] 
[cmake]   was specified.
```

I looked in `cmake-tools-kits.json` , `user settings.json`, `workspace settings.json` and I haven't declared anywhere the platform. However, I can see that in the `CMakeCache.txt` it is set `CMAKE_GENERATOR_PLATFORM:INTERNAL=x64` . I can't figure out where it comes form, anyone has any ideas?

**Comment by calebkiage** [comment]
Maybe you have a [toolchain](https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html) file that sets it or a [preset](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html).
The cmake configure preset's `architecture` sets the `CMAKE_GENERATOR_PLATFORM` cache variable.

**Comment by bobbrow** [comment]
Can you set `cmake.loggingLevel` to `"debug"` and re-run the configure step?  I'd like to see what the cmake command is that we're generating.  Search for a string like `[proc] Executing command:` in the Output Window after running the `CMake: Configure` command.

**Reply by George-Gi** [reply]
I solved the issue by closing the VSCode, deleting entirely the build folder generated by CMake, and doing configuration again. I still haven't figured out where it was getting the -A flag from.
I did the run under debug log:

```
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.exe" --build c:/Users/userName/Project/build --config Debug --target Project_test --
```

**Reply by gcampbell-msft** [reply]
@George-Gi Looks like your issue was solved here. I don't have enough context to investigate, but if you're still hitting issues with this please feel free to create a GitHub issue and we'd be happy to investigate. Thanks!
